### DISCUSSION TITLE
v1.5.3 to use `SPI_DMA_CH_AUTO` and `built-in ESP32 MAC`

### DISCUSSION THREAD

**Discussion by khoih-prog** [khoih-prog/WebServer_ESP32_W5500#5] [Announcements]
#### Releases v1.5.3

1. Using `SPI_DMA_CH_AUTO` and **built-in ESP32 MAC**
2. Improve library and examples code

<hr /><em>This discussion was created from the release <a href='https://github.com/khoih-prog/WebServer_ESP32_W5500/releases/tag/v1.5.3'>v1.5.3 to use `SPI_DMA_CH_AUTO` and `built-in ESP32 MAC`</a>.</em>

**Comment by drwho100** [comment]
Chào anh, em đang học esp32 và w5500. theo em tìm tài liệu thì 1 số lib w5500 không cần chân INT nhưng khi em sử dụng lib của anh thì bị bắt buộc sử dụng chân INT cho GPIO4 . vậy mục đích của chân này để làm gì và nếu em muốn sử dụng thư viện của anh thì làm thế nào để bỏ qua chân INT này ? Cảm ơn anh hướng dẫn.

**Reply by khoih-prog** [reply]
You can use [**Ethernet_Generic**](https://github.com/khoih-prog/Ethernet_Generic) and [**EthernetWebServer**](https://github.com/khoih-prog/EthernetWebServer) libraries if don't want to use INT. These libraries only can be used only in Sync Mode.

This new library is designed for better performance and must be used if Async mode is requested.

Try to read the library's code and interrupt ISR function, and research why using INT is better.

https://github.com/khoih-prog/WebServer_ESP32_W5500/blob/05494bfc3c5c0a28bb1dc330dfd856f5c6b5c97a/src/w5500/esp_eth/esp_eth_mac_w5500.c#L830

https://github.com/khoih-prog/WebServer_ESP32_W5500/blob/05494bfc3c5c0a28bb1dc330dfd856f5c6b5c97a/src/w5500/esp_eth/esp_eth_mac_w5500.c#L426-L438

Hope this helps,
